### PR TITLE
Wrapper class for parsed trace stats.

### PR DESCRIPTION
**Pull request by mmichal10**
Signed-off-by: Michal Mielewczyk <michal.mielewczyk@intel.com>

### CODE REVIEW

File: tests/api/iotrace_parser.py [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+    def __str__(self):
+        ret = f"metrics {self.metrics}"
+        ret += f"errors {self.errors}\n"
+
+        return ret
+
+
+class Percentiles:
+    def __init__(self, trace, message="of population have value lower than"):
+        TestRun.LOGGER.info(f"{trace}")
```
Comment (tomaszrybicki) [inline tests/api/iotrace_parser.py]: Why are we printing here?

File: tests/api/iotrace_parser.py [hunk 2]
```diff
@@ -0,0 +1,173 @@
+#
```
Comment (tomaszrybicki) [inline tests/api/iotrace_parser.py]: I would name the parser something like iotrace_stats_parser.py
Unless we will put put parsing of every iotrace command here later on

File: tests/api/iotrace_parser.py (hunk 1 above)
Comment (mmichal10) [reply tests/api/iotrace_parser.py]: It was printed for debug purposes. I removed it in latest version.

File: tests/api/iotrace_stats_parser.py [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+        self.percentiles = Percentiles(
+            trace["percentiles"], "of reqs have latency lower than"
+        )
+
+    def __str__(self):
+        ret = f"average latency: {self.average} ns\n"
+        ret += f"min latency: {self.min} ns\n"
+        ret += f"max latency: {self.max} ns\n"
+        ret += f"count: {self.count}\n"
+        ret += f"total: {self.min} ns\n"
```
Comment (Ostrokrzew) [inline tests/api/iotrace_stats_parser.py]: I think You wanna write `self.total`.

File: tests/api/iotrace_stats_parser.py [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+        self.total = IOStatistics(trace["total"])
+
+    def __str__(self):
+        ret = f"{self.device}"
+        ret += f"duration: {self.duration}\n"
+        ret += f"duration: {self.read}"
+        ret += f"duration: {self.write}"
+        ret += f"duration: {self.discard}"
+        ret += f"duration: {self.flush}"
+        ret += f"duration: {self.total}"
```
Comment (Ostrokrzew) [inline tests/api/iotrace_stats_parser.py]: Should there be repeated word 'duration'?

File: tests/api/iotrace_stats_parser.py [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+
+class DeviceStats:
+    def __init__(self, trace):
+        self.id = trace["id"]
+        self.name = trace["name"]
+        self.size = Size(float(trace["size"]), Unit.Blocks512)
+
+    def __str__(self):
+        ret = f"device: id: {self.id}, name: {self.name}, size {self.size}\n"
+        return ret
```
Comment (Ostrokrzew) [inline tests/api/iotrace_stats_parser.py]: You don't have to keep this string in variable. You could pass this directly to return.

File: tests/api/iotrace_stats_parser.py [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+        self.percentiles = Percentiles(
+            trace["percentiles"], "of reqs touches no more sectors than"
+        )
+
+    def __str__(self):
+        ret = f"average size: {self.average} sector(s)\n"
+        ret += f"min size: {self.min} sector(s)\n"
+        ret += f"max size: {self.max} sector(s)\n"
+        ret += f"count: {self.count}\n"
+        ret += f"total: {self.min} sector(s)\n"
```
Comment (Ostrokrzew) [inline tests/api/iotrace_stats_parser.py]: I think You wanna write self.total.

**Comment by tomaszrybicki** [comment]
retest this please

**Pull request merged**